feat: add user profile

Abas-dev opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

​

## Description

<!--- Describe your changes in detail -->

​I worked on the creation of user profile using the users token and to get the user that is currntly logged in using the token.

## Related Issue (https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/438)

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

​

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
This endpoint will create a user profile using the provided token and if a user already have an account, it through an error.
​

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

​

## Screenshots (if appropriate - Postman, etc):
![Screenshot (168)](https://github.com/user-attachments/assets/3b278416-3aae-41cd-b8ae-3c57d51e149a)




​

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Abas-dev commented:
@zxenonx 

pls check

zxenonx commented:
> @zxenonx
> 
> pls check

Please follow the response format.

Abas-dev commented:
> > @zxenonx
> > pls check
> 
> Please follow the response format.

i have corrected it boss

zxenonx commented:
> > > @zxenonx
> > > pls check
> > 
> > 
> > Please follow the response format.
> 
> i have corrected it boss @zxenonx

You need to update the screenshot then.

Abas-dev commented:
I have updated the screenshot

On Fri, 26 Jul 2024, 12:21 pm zxenonx, ***@***.***> wrote:

> @zxenonx <https://github.com/zxenonx>
> pls check
>
> Please follow the response format.
>
> i have corrected it boss @zxenonx <https://github.com/zxenonx>
>
> You need to update the screenshot then.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/pull/468#issuecomment-2252549077>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS5EYWFLDSZLNLRNYN4XITTZOIWK7AVCNFSM6AAAAABLOSC5XOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENJSGU2DSMBXG4>
> .
> You are receiving this because you authored the thread.Message ID:
> <hngprojects/hng_boilerplate_python_fastapi_web/pull/468/c2252549077@
> github.com>
>

zxenonx commented:
> I have updated the screenshot
> […](#)
> On Fri, 26 Jul 2024, 12:21 pm zxenonx, ***@***.***> wrote: @zxenonx <https://github.com/zxenonx> pls check Please follow the response format. i have corrected it boss @zxenonx <https://github.com/zxenonx> You need to update the screenshot then. — Reply to this email directly, view it on GitHub <[#468 (comment)](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/pull/468#issuecomment-2252549077)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AS5EYWFLDSZLNLRNYN4XITTZOIWK7AVCNFSM6AAAAABLOSC5XOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENJSGU2DSMBXG4> . You are receiving this because you authored the thread.Message ID: <hngprojects/hng_boilerplate_python_fastapi_web/pull/468/c2252549077@ github.com>

Nothing has changed then - the response body format is still wrong

Abas-dev commented:
Ah 


That's the correct response body format 

It's to return the data of the created profile and the id of the user

zxenonx commented:
> Ah
> 
> That's the correct response body format
> 
> It's to return the data of the created profile and the id of the user

Ask your teammates.

Abas-dev commented:
fixed

joboy-dev reviewed (changes requested):
Resolve conflicts

Abas-dev commented:
@Goketech @joboy-dev 

merge
